### PR TITLE
Feature remote STATIC extensions

### PR DESCRIPTION
**Pull request by huboneo**
### Please check if the PR fulfills these requirements
- [x] The commit message follows our [guidelines](https://github.com/neo4j-devtools/relate/blob/master/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce?
This PR adds support for using remote apps, defined as an installed STATIC extension whose `main` property contains a URL.

This PR also updates various URL parameters used by relate to match those of Neo4j Desktop, and exposes a new client utility for grabbing them from URLs.

Further this PR changes the GraphQL schema for apps to include a `url` property instead of a `path`.

### What is the current behavior?
Can only open locally installed apps

### What is the new behavior?
Apps can point to a URL

### Does this PR introduce a breaking change?
yes, this PR changes the GraphQL schema for apps to include a `url` property instead of a `path`.

### Other information:
Sample remote app:
```json
{
  "name": "gewgle",
  "version": "1.0.1",
  "main": "https://google.com"
}
```
[extension-gewgle.zip](https://github.com/neo4j-devtools/relate/files/5458309/extension-gewgle.zip)

CLI Usage
```
$ relate app:open gewgle -L
> https://google.com/?relateUrl=http%3A%2F%2F127.0.0.1%3A3000
```

WEB Usage
Screenshot 2020-10-29 at 13 32 57

### CODE REVIEW

File: packages/common/src/utils/extensions/get-app-launch-url.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    launchToken?: string,
+): Promise<string> {
+    const asURL = isValidUrl(appPath) ? new URL(appPath) : new URL(`${environment.httpOrigin}${appPath}`);
+
+    if (asURL.origin !== environment.httpOrigin) {
+        asURL.searchParams.set('relateUrl', environment.httpOrigin);
+    }
+
+    if (environment.requiresAPIToken) {
+        asURL.searchParams.set('relateApiToken', await environment.generateAPIToken(asURL.host, clientId, {}));
```
Comment (huboneo) [inline packages/common/src/utils/extensions/get-app-launch-url.ts:17]: There is an assumption here, that appName is essentially the clientID. I dont know how we could enforce/communicate this convention to outside devs

File: packages/common/src/utils/extensions/get-app-launch-url.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+export async function getAppLaunchUrl(
+    environment: EnvironmentAbstract,
+    appPath: string,
+    clientId: string,
+    launchToken?: string,
+): Promise<string> {
+    const asURL = isValidUrl(appPath) ? new URL(appPath) : new URL(`${environment.httpOrigin}${appPath}`);
+
+    if (asURL.origin !== environment.httpOrigin) {
+        asURL.searchParams.set('relateUrl', environment.httpOrigin);
```
Comment (nglgzz) [inline packages/common/src/utils/extensions/get-app-launch-url.ts:13]: Shouldn't we use constants for these parameters?

Comment (huboneo) [reply packages/common/src/utils/extensions/get-app-launch-url.ts:13]: I'll add it in, but not sure who would use them?

**Review by nglgzz — APPROVED** [review]
Looks good to me

File: packages/common/src/utils/extensions/get-app-launch-url.ts (hunk 2 above)
Comment (huboneo) [reply packages/common/src/utils/extensions/get-app-launch-url.ts]: but yeah, good call :)

**Pull request merged**